src/west/app/project.py

```diff
@@ -556,14 +556,22 @@ def do_run(self, args, ignored):
         self._setup_logging(args)
 
         failed = []
+        # We may need to force git to use colors if the user wants them,
+        # which it won't do ordinarily since stdout is not a terminal.
+        color = ['--color=always'] if log.use_color() else []
+
         for project in self._cloned_projects(args):
-            log.banner(f'diff for {project.name_and_path}:')
             # Use paths that are relative to the base directory to make it
             # easier to see where the changes are
-            try:
-                project.git(['diff', f'--src-prefix={project.path}/',
-                             f'--dst-prefix={project.path}/'])
-            except subprocess.CalledProcessError:
+            cp = project.git(['diff', f'--src-prefix={project.path}/',
+                              f'--dst-prefix={project.path}/',
+                              '--exit-code'] + color,
```

Comment:
BTW, I tried to do the same thing for 'git status', but ran into a potential problem. The `--porcelain=2` option is not available on Ubuntu 16.04 and I believe that was required to get machine-readable output from `git status` which can be colorized. I think we need to either:

- cache the git version in the WestApp and pass it to the WestCommand, which can check for features with something like `if self.git_version >= SOME_MIN_VERSION: # use git status --porcelain=2`, to try to keep supporting older versions of git, OR
- start tracking a minimum version of git that west's core supports and testing against it and only use features that are always available in the minimum version

This gets tricky and is a general problem, IMO.